### PR TITLE
Database migrations to allow school dropdown to be added to teacher registration

### PR DESCRIPTION
**Pull request by ewjordan**
- Adds a school_info reference to the users table
- Adds validation_type string to school_infos table (for now, intended to be either 'full' or 'none' - we may have an intermediate value if we decide to constrain the allowed partially filled states a bit more)

These migrations are in preparation to add the school dropdown to a percentage of new teacher signups, and allow all fields to be optional when set from there (but not in other places where we supply school_info, which is why we need the ability to suppress the usual validation strategy).

This needs to be deployed before the school dropdown changes go in.

### CODE REVIEW

File: dashboard/db/migrate/20170207004002_add_validation_type_to_school_info.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddValidationTypeToSchoolInfo < ActiveRecord::Migration[5.0]
+  def change
+    add_column :school_infos, :validation_type, :string, default: 'full'
```
Comment (aoby) [inline dashboard/db/migrate/20170207004002_add_validation_type_to_school_info.rb]: You might want to add `null: false` too, to make sure this is always set.

File: dashboard/db/migrate/20170202090015_add_school_info_to_user.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddSchoolInfoToUser < ActiveRecord::Migration[5.0]
+  def change
+    add_reference :users, :school_info, index: true
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20170202090015_add_school_info_to_user.rb]: Does it make sense to set a default value of NULL?

Comment (ashercodeorg) [reply dashboard/db/migrate/20170202090015_add_school_info_to_user.rb]: Does it make sense to put this column after the existing `school` column?

File: dashboard/app/models/school_info.rb [hunk 3]
```diff
@@ -16,6 +16,7 @@
 #  full_address          :string(255)
 #  created_at            :datetime         not null
 #  updated_at            :datetime         not null
+#  validation_type       :string(255)      default("full")
```
Comment (ashercodeorg) [inline dashboard/app/models/school_info.rb]: Ideally, after this PR DTPs, another PR adds a validation to the `validation_type` column confirming its value is from an enum list.

**Comment by ashercodeorg** [comment]
LGTM (see comments).

**Comment by ewjordan** [comment]
@ashercodeorg updated, mind taking another quick look?

**Comment by ashercodeorg** [comment]
LGTM.

**Pull request merged**